identity-core/src/common/timestamp.rs

```diff
@@ -165,6 +188,49 @@ impl Diff for Timestamp {
   }
 }
 
+/// A span of time.
+///
+/// This type is typically used to increment or decrement a [`Timestamp`].
+#[derive(Clone, Copy, Hash, PartialEq, Eq, PartialOrd, Ord, Deserialize, Serialize)]
+#[repr(transparent)]
+pub struct Duration(time::Duration);
+
+// copy a small subset of the functionality offered by time::Duration. By not copying everything it will be easier
+// to migrate to another implementation if that will be necessary in the future.
+// Do not allow negative Durations hence all constructors take u32 instead of i64, this makes it possible for us to
+// switch to relying on (for instance) std::time::Duration if we want to.
+impl Duration {
+  /// A second
+  pub const SECOND: Self = Self(time::Duration::SECOND);
+  /// A minute
+  pub const MINUTE: Self = Self(time::Duration::MINUTE);
+  /// An hour
+  pub const HOUR: Self = Self(time::Duration::HOUR);
+  /// A day
+  pub const DAY: Self = Self(time::Duration::DAY);
+  /// A week
+  pub const WEEK: Self = Self(time::Duration::WEEK);
```

Comment:
Do we need to expose these constants? There's no way to compose them so they're not very useful right now. 

I.e. we do not implement arithmetic operators, so it's not possible to do something like `Duration::WEEK + 3*Duration::DAY + 5*Duration::MINUTE`. This means developers will need to convert any desired duration to the smallest unit, e.g. `Duration::minutes(7*24*60 + 3*24*60 + 5)`. Not great but at least there's a way I guess. 

We could also expose something like `Duration::new(weeks, days, hours, minutes, seconds)`. Not sure which direction to go really, but the point is that without arithmetic operations these constants are not useful in my opinion.

Reply:
Minimal API is preferable at this point so I will simply remove them. We can add `Duration::new(weeks, days, hours, minutes, seconds)` at a later point if someone requests it.